docs/pair_hmm.tex

```diff
@@ -0,0 +1,101 @@
+
+\begin{equation}
+P(\mc{R} | \mc{H}) = \sum_{\mc{A}} P(\mc{R}, \mc{A} | \mc{H})
+\end{equation}
+
+We represent alignments as sequences
+\begin{equation}
+{\rm alignment} = \{ (i_1, j_1, s_1), (i_2, j_2, s_2) \ldots (i_N, j_N, s_N) \},
+\end{equation}
+where $i$ and $j$ represent positions within the read and haplotype, respectively, and $s_n \in \{ M, I, D \}$ represents the states of match, insertion, and deletion of the read relative to the haplotype.  For example, an alignment $\{ (1,10, M), (2,11, M), (3,12, M), (4,13, M), (4,14, D), (4,15, D), (5,16, M), (6,17, M), (7,17, I), (8,18, M) \}$ means that positions 1 - 4 of the read match positions 10 - 13 of the haplotype, followed by a two-base deletion (advancing from 13 to 15 in the haplotype without advancing in the read), followed by a match of read positions 5 - 6 with haplotype positions 16 - 17, followed by an insertion at read position 7, followed by a match.  The allowable transitions $(i_n, j_n) \rightarrow (i_{n+1}, j_{n+1})$ are
+\begin{itemize}
+\item $(i, j, M/D/I) \rightarrow (i+1, j+1, M)$: match of read position $i+1$ with haplotype position $j + 1$
+\item $(i, j, M/D) \rightarrow (i, j+1, D)$: deletion after read position $i$ -- haplotype position $j+1$ is deleted
+\item $(i, j, M/I) \rightarrow (i + 1, j, I)$: insertion at read position $i + 1$
+\end{itemize}
+Note that the state label $s$ seems redundant because it can be reconstructed from the sequence of $i$ an $j$.  While this is true, our model treats indel starts differently from indel continuations, and by including the state label we can distinguish these conveniently\footnote{That is, we can tell which type of transition it is by looking back one unit instead of two.  Thus we have a first-order Markov model instead of a second-order Markov model.}.
+
+The read-alignment likelihood has two components.  First is the probability of the sequence of match, insertion, and deletion states, which is
+\begin{equation}
+P(\mc{A}) = \prod_k T_{s_k, s_{k+1}},
+\end{equation}
+where $T$ is a constant matrix of state transition probabilities\footnote{The gap continuation probability corresponds to a phred-scaled quality that is set by the \code{gcpHMM} argument, which is 10 by default, implying that $T_{DD} = T_{II} = 0.1$ and $T_{DM} = T{IM} = 0.9$.  The other transitions are hard-coded based on a phred-scaled indel start quality of 45.  That is, $T_{MD} = T{MI} = 10^{-4.5}$ and $T_{MM} = 1 - 2 \times 10^{-4.5}$.  This fact is somewhat obscured in the code, which gives a position index to the transition matrices.  Nonetheless, the array representing transitions is constant over this index.}.  Next is the emission probability of the read bases given the haplotype bases they align to and the base qualities:
```

Comment:
Agreed.  We don't do insertion or deletion qualities in production because they make the bams too big.